### PR TITLE
Minor refactoring - replaced deprecated methods around `dumbSlave()`

### PR DESCRIPTION
**Pull request by StefanSpieker**
Minor refactoring:
- replaced deprecated methods around `dumbSlave()`
- use `TemporaryFolder` instead of dreprecated `JenkinsRule.createTmpDir()` as suggested

### Proposed changelog entries

- N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7242"></a>

### CODE REVIEW

File: test/src/test/java/hudson/model/labels/LabelExpressionTest.java
Comment (basil) [inline test/src/test/java/hudson/model/labels/LabelExpressionTest.java:216]: Prefer immutable lists to mutable lists where possible:

```suggestion
        slave.setNodeProperties(List.of());
```

**Review by basil — COMMENTED** [review]
Usage of `org.junit.rules.TemporaryFolder` looks wrong: while `#toString` might work, the Javadoc shows that the expected typical use case is to call either `#newFile` or `#newFolder`.

File: test/src/test/java/hudson/model/labels/LabelExpressionTest.java
Comment (StefanSpieker) [reply test/src/test/java/hudson/model/labels/LabelExpressionTest.java]: I used the `Collections.EMPTY_LIST` as before. I guess this is also fine, because it is also immutable. But thanks for the hint.

**Comment by StefanSpieker** [comment]
I now correctly use the Rule, thanks @basil

File: test/src/test/java/hudson/model/FreeStyleProjectTest.java
Comment (basil) [inline test/src/test/java/hudson/model/FreeStyleProjectTest.java:119]: No need to convert the first argument to a string:

```suggestion
        File d = new File(tempFolder.newFolder(), "${JOB_NAME}");
```

File: test/src/test/java/hudson/model/labels/LabelExpressionTest.java
Comment (basil) [inline test/src/test/java/hudson/model/labels/LabelExpressionTest.java:216]: Per the Javadoc, the field does not prefer type safety. The type-safe way to obtain an empty list is:

```suggestion
        slave.setNodeProperties(Collections.emptyList());
```

**Review by basil — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Review by NotMyFault — APPROVED** [review]

**Comment by StefanSpieker** [comment]
@basil Thanks a lot for solving merge conflicts. Very appreciated.

File: test/src/test/java/hudson/model/labels/LabelExpressionTest.java
Comment (jglick) [inline test/src/test/java/hudson/model/labels/LabelExpressionTest.java:216]: FTR one of the reasons for introducing the simplified constructor to begin with was to allow sensible defaults for most properties. At least `nodeDescription` and `nodeProperties` need not be overridden here, and the same is probably true of `retentionStrategy`.

**Pull request merged**
